drivers/clock_control/clock_stm32_ll_u5.c

```diff
@@ -0,0 +1,607 @@
+__unused
+static void set_up_clk_msis(void)
+{
+	/* Set MSIS Range */
+	LL_RCC_MSI_EnableRangeSelection();
+
+	LL_RCC_MSIS_SetRange(STM32_MSIS_RANGE << RCC_ICSCR1_MSISRANGE_Pos);
+
+	if (STM32_MSIS_RANGE < 4) {
+		/* MSI clock trimming for ranges 0 to 3 */
+		LL_RCC_MSI_SetCalibTrimming(0, LL_RCC_MSI_OSCILLATOR_0);
+	} else if (STM32_MSIS_RANGE < 8) {
+		/* MSI clock trimming for ranges 4 to 7 */
+		LL_RCC_MSI_SetCalibTrimming(0, LL_RCC_MSI_OSCILLATOR_1);
+	} else if (STM32_MSIS_RANGE < 12) {
+		/* MSI clock trimming for ranges 8 to 11 */
+		LL_RCC_MSI_SetCalibTrimming(0, LL_RCC_MSI_OSCILLATOR_2);
+	} else {
+		/* MSI clock trimming for ranges 12 to 15 */
+		LL_RCC_MSI_SetCalibTrimming(0, LL_RCC_MSI_OSCILLATOR_3);
+	}
+
+#if STM32_MSIS_PLL_MODE
+
+#if !STM32_LSE_CLOCK
+#error "MSI Hardware auto calibration requires LSE clock activation"
+#endif
+	/* Enable MSI hardware auto calibration */
+	LL_RCC_MSI_EnablePLLMode();
+#endif
+
+	/* Set MSIS Range */
+	LL_RCC_MSIS_Enable();
+
+	/* Wait till MSIS is ready */
+	while (LL_RCC_MSIS_IsReady() != 1) {
+	}
+}
+#endif /* STM32_SYSCLK_SRC_MSIS || STM32_PLL_SRC_MSIS */
+
+#if STM32_SYSCLK_SRC_PLL
+/*
+ * Configure PLL as source of SYSCLK
+ */
+void config_src_sysclk_pll(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
+{
+	LL_UTILS_PLLInitTypeDef s_PLLInitStruct;
+
+	/* configure PLL input settings */
+	config_pll_init(&s_PLLInitStruct);
+
+	/*
+	 * Switch to HSI and disable the PLL before configuration.
+	 * (Switching to HSI makes sure we have a SYSCLK source in
+	 * case we're currently running from the PLL we're about to
+	 * turn off and reconfigure.)
+	 *
+	 * Don't use s_ClkInitStruct.AHBCLKDivider as the AHB
+	 * prescaler here. In this configuration, that's the value to
+	 * use when the SYSCLK source is the PLL, not HSI.
+	 */
+	clock_switch_to_hsi(LL_RCC_SYSCLK_DIV_1);
+	LL_RCC_PLL1_Disable();
+
+#if STM32_PLL_Q_DIVISOR
+	LL_RCC_PLL1_SetQ(STM32_PLL_Q_DIVISOR);
+#endif /* STM32_PLL_Q_DIVISOR */
+
+#if STM32_PLL_SRC_MSIS
+
+	/*
+	 * For now, an issue detected in function LL_PLL_ConfigSystemClock_MSI
+	 * doesn't allow it's use in current Cube package version (1.0.0).
+	 * So we're using step by step configuration, with fixed flash latency
+	 * setting.
+	 * This has been tested using max supported freq (160MHz), but could
+	 * have limitations in lower speed settings.
+	 */
+
+	LL_FLASH_SetLatency(LL_FLASH_LATENCY_4);
+	while (LL_FLASH_GetLatency() != LL_FLASH_LATENCY_4) {
+	}
+
+	set_up_clk_msis();
+
+	LL_RCC_PLL1_ConfigDomain_SYS(LL_RCC_PLL1SOURCE_MSIS,
+				     STM32_PLL_M_DIVISOR,
+				     STM32_PLL_N_MULTIPLIER,
+				     STM32_PLL_R_DIVISOR);
+
+	if (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC >= 55) {
+		/*
+		 * Set EPOD prescaler based on PLL1 input freq (MSI/PLLM)
+		 * Booster clock frequency should be between 4 and 16MHz
+		 * This is done in following steps:
+		 * Read MSI Frequency
+		 * Didvide PLL1 input freq (MSI/PLLM) by the targetted freq (8MHz)
+		 * Make sure value is not higher than 16Mhz
```

Comment:
```suggestion
		 * Make sure prescaler value is not higher than 16
```
This is prescaler value it has no unit